genai_stack/vectordb/weaviate_db/weaviate.py

```diff
@@ -51,6 +52,45 @@ def _create_langchain_client(self, **kwargs):
             client=self.client, embedding=self.mediator.get_embedding_function(), by_text=False, **kwargs
         )
 
+    def hybrid_search(
+        self,
+        query: str,
+        metadata: dict,
+        k=1,
+        **kwargs,
+    ):
+        if kwargs.get("attributes", None):
+            kwargs['attributes'] += self.config.config_data.attributes
+        args = {
+            "query": query,
+            "k": k
+        }
+        if metadata:
+            where_filter = {
+                "operator": "And",
+                "operands": [
+                    {
+                        "path": [key],
+                        "operator": "Equal",
+                        "valueString": value,
+                    } if type(value) == str else {
+                        "path": [key],
+                        "operator": "Equal",
+                        "valueNumber": value,
+                    } for key, value in metadata.items()
+                ]
+            }
+            args["where_filter"] = where_filter
+        lc_client = self._create_langchain_client(**kwargs)
+        documents = lc_client.similarity_search_with_score(**args)
+        return [{
+            "query": document[0].page_content,
```

Comment:
Create a pydantic model for this instead of passing the key value pairs manually . It would be consistent with our architecture also.